site_scons/stack_analyzer.py

```diff
@@ -79,15 +79,15 @@ def analyze(self):
         """Run the analysis"""
         function_map = {}
 
-        print("Analyzing %s" % self.base_dir)
-        print("Options:")
-        print("\texcluded directory strings: %s" % self.get_value(self.dir_exclusions, "none"))
-        print("\tincluded directory strings: %s" % self.get_value(self.dir_inclusions, "all"))
-        print("\tincluded file strings     : %s" % self.get_value(self.file_inclusions, "all"))
-        print("\tcutoff                    : %d" % self.cutoff)
+        print(f'"Analyzing {self.base_dir}')
```

Comment:
Not a python expert, but what is the criteria for using f' versus no f' in these print statements?

Reply:
Everything except ftest code should be using f-strings now.

Reply:
If you're asking what `f'` is for, it's python 3.6+ formatting. I.e., these two lines are equivalent:
```
'my_var = {}'.format(my_var)
f'my_var = {my_var}'
```

Reply:
I was curious why this line used print(f'text {var}')
and some of the lines immediately following do not use the f' (e.g., print('\tcutoff  : {self.cutoff}')

Reply:
We have many hundred detected python warnings in the codebase, some of them we've fixed directly in PRs but there are too many for this to be viable for all, so for others we ask that users resolve them when working on the code, github and the commit-hook will warn you about existing warnings within a few lines of code that you've modified.